### PR TITLE
cli new ruleset

### LINKED ISSUE #35: CLI - create new ruleset
Some type of command to initialize / create a new ruleset project

### PR DESCRIPTION
**Pull request by BaderBC**
Closes #35

### CODE REVIEW

**Review by cooligus — CHANGES_REQUESTED** [review]
Implement adding lua file with empty main

**Comment by BaderBC** [comment]
> Implement adding lua file with empty main

Do you mean creating `example_rule.lua` in order to show how our rule syntax looks like?

File: src/cmd/new/new.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+//go:embed "initial_ruleset/spito.yaml"
+var spitoYaml []byte
+
+func newRuleset(name string) error {
+	if err := os.Mkdir(name, os.ModePerm); err != nil {
+		return err
+	}
+
+	return os.WriteFile(name+"/spito.yaml", spitoYaml, os.ModePerm)
```
Comment (sewe2000) [inline src/cmd/new/new.go:32]: Where exactly will this spito.yaml file be?

Comment (BaderBC) [reply src/cmd/new/new.go:32]: We discussed in the last meeting that `spito-rules.yaml` file, which you can find in [this](https://github.com/Nasz-Elektryk/spito-ruleset/) repo, will be renamed to `spito.yaml`

Comment (sewe2000) [reply src/cmd/new/new.go:32]: OK, I understand. I humbly propose to rename this `name` variable something like `directoryName` to avoid confusion.

**Comment by cooligus** [comment]
> Do you mean creating `example_rule.lua` in order to show how our rule syntax looks like?

No. I mean to create lua file and link it to spito.yaml.

**Comment by BaderBC** [comment]
> > Do you mean creating `example_rule.lua` in order to show how our rule syntax looks like?
> 
> No. I mean to create lua file and link it to spito.yaml.

Okay but what name do you propose for that lua file `example.lua`?

**Comment by cooligus** [comment]
> Okay but what name do you propose for that lua file `example.lua`?

Name is specified in command

**Comment by BaderBC** [comment]
> Name would be specified in command

But the name specified in the command is the name of the ruleset not particular rule

**Comment by cooligus** [comment]
So I propose this syntax:
```
spito new rules-set neovim gnome
```

or split this into two commands:
```
spito new set name-of-set
spito new rule name-of-rule
```

**Comment by BaderBC** [comment]
What about:
```
spito new name-of-set
spito (g|generate) rule name-of-rule
```

Where `spito new name-of-set` create also rule `example.lua` with hello world inside it (by using `api.info.Log`)

**Comment by cooligus** [comment]
Sounds good, but instead of `example.lua` i suggest to name it `name-of-rule.lua`

**Review by sewe2000 — APPROVED** [review]

**Comment by cooligus** [comment]
There is no `spito generate` and I meant `name-of-rule` to be variable obtained from user input

**Comment by BaderBC** [comment]
@sewe2000 misunderstood his task and instead of extending this pr he created its own (#47)
That's why I will close this pr

**Pull request closed without merging**